pkg/util/http.go

```diff
@@ -374,3 +384,43 @@ func copyValues(src url.Values) url.Values {
 func IsHTTPStatusCode(code codes.Code) bool {
 	return int(code) >= 100 && int(code) < 600
 }
+
+// RequestBuffers provides pooled request buffers.
+type RequestBuffers struct {
+	p       *sync.Pool
+	buffers []*bytes.Buffer
+	// Allows avoiding heap allocation
+	buffersBacking [10]*bytes.Buffer
+}
+
+// NewRequestBuffers returns a new RequestBuffers given a sync.Pool.
+func NewRequestBuffers(p *sync.Pool) *RequestBuffers {
+	rb := &RequestBuffers{
+		p: p,
+	}
+	rb.buffers = rb.buffersBacking[:0]
+	return rb
+}
+
+// Buffer obtains a buffer from the pool.
```

Comment:
I'd suggest renaming `Buffer` to `Get` or `GetBuffer`.

Reply:
The reason why I named it `Buffer` is the Go convention of not prefixing method names with `Get`. I guess just `Get` makes sense though, as it mirrors `Pool.Get`.

Reply:
Done.